types/MediaConnector.d.ts

```diff
@@ -1,39 +1,44 @@
 import { QueryOptions, Dictionary, ArrayBufferPointer, DownloadType } from "./Connector.Shared";
 
-declare module "grafx-studio-mediaconnector" {
+export interface MediaConnector {
+    query(options: QueryOptions, context: Dictionary): Promise<MediaPage>;
+    detail(id: string): Promise<MediaDetail>;
+    download(id: string, previewType: DownloadType, context: Dictionary): Promise<ArrayBufferPointer>
+    upload(name: string, blob: ArrayBufferPointer): Promise<Media>
+    remove(id: string): Promise<boolean>
+    copy(id: string, newName: string): Promise<Media>
+    getQueryOptions(): string[] | null;
+    getDownloadOptions(): string[] | null;
+    getCapabilities(): MediaConnectorCapabilities;
+}
 
-    interface MediaConnector {
-        query(options: QueryOptions, context: Dictionary): Promise<MediaPage>;
-        download(id: string, previewType: DownloadType, context: Dictionary): Promise<ArrayBufferPointer>
-        upload(name: string, blob: Int8Array): Promise<Media>
-        remove(id: string): Promise<boolean>
-        copy(id: string, newName: string): Promise<Media>
-        getQueryOptions(): string[] | null;
-        getDownloadOptions(): string[] | null;
-        getCapabilities(): MediaConnectorCapabilities;
-    }
-    
-    type MediaConnectorCapabilities = {
-        filtering: boolean;
-        upload: boolean;
-        query: boolean;
-        remove: boolean;
-        copy: boolean;
-    }
+export type MediaConnectorCapabilities = {
+    filtering: boolean;
+    upload: boolean;
+    query: boolean;
+    detail: boolean;
+    remove: boolean;
+    copy: boolean;
+}
 
-    interface MediaPage {
-        pageSize: number;
-        data: Media[];
-        links: {
-            nextPage: string;
-        };
-    }
+export interface MediaPage {
+    pageSize: number;
+    data: Media[];
+    links: {
+        nextPage: string;
+    };
+}
 
-    interface Media {
-        id: String;
-        name: String;
-        relativePath: String;
-        type: number;
-        metaData: Dictionary;
-    }
+export interface Media {
+    id: String;
```

Comment:
any particular reason to use String vs string?

Reply:
goowd one 🦅👁️

Reply:
Updated the PR to use the Typescript native `string` everywhere in the type definitions (those were the only occurences)